### ISSUE TITLE
SwitchPreference - odd behaviour

### ISSUE THREAD

**Issue by indieboyalex** [Prototik/HoloEverywhere#90]
Hey!

Just found an odd behaviour with the SwitchPreference.

I have an preferences.xml file with:
1. SwitchPreference
2. SwitchPreference - Controls enabled/disabled state of RingtonePreference
3. RingtonePreference

Now, when I attempt to disable switch 2 (and disable 3), it always flicks back on straight away.  However when I turn switch 1 off it turns switch 2 off and then disables 3.

Obviously this isn't quite what we want....

I'll just paste in the relevant bit of my XML, you probably remember the rest of it from last week anyway :)

```
   <SwitchPreference
        holo:defaultValue="false"
        holo:key="adult"
        holo:persistent="true"
        holo:summary="Show content marked at 18+"
        holo:title="Adult Content" />
</PreferenceCategory>

<PreferenceCategory
    holo:key="pref_category_notificaions"
    holo:title="@string/notifications" >

    <SwitchPreference
        holo:defaultValue="true"
        holo:key="notif_active"
        holo:persistent="true"
        holo:title="Notifications Active" />

    <RingtonePreference
        holo:dependency="notif_active"
        holo:key="notif_sound"
        holo:persistent="true"
        holo:ringtoneType="notification"
        holo:showDefault="true"
        holo:title="Notification Sound" >
    </RingtonePreference>

</PreferenceCategory>
```

**Comment by Prototik** [comment]
Tommorow, okay?
I'm already sleeeeep....

**Comment by indieboyalex** [comment]
That's fine - I'll be away from work till Monday.

Sleep well :)

**Comment by Prototik** [comment]
It... it's bug in android framework.
PreferenceGroupAdapter.
getItemViewType.
Has:

```
        if (preference.hasSpecifiedLayout()) {
            return IGNORE_ITEM_VIEW_TYPE;
        }
```

Should:

```
        if (!preference.hasSpecifiedLayout()) {
            return IGNORE_ITEM_VIEW_TYPE;
        }
```

Will be fixed in 1.2.4

**Comment by indieboyalex** [comment]
Thanks :)

Though I did have fun flicking an entire page on and off at the same time.
Simple pleasures, eh!

Jeff
